controllers/subscription.go

```diff
@@ -22,44 +22,43 @@ import (
 	"log"
 	"reflect"
 
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
 
-	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+func newSubscriptionFromConfigMap(r kubernetes.Interface) (*operatorv1alpha1.Subscription, error) {
+	var newSubscription *operatorv1alpha1.Subscription
 
-	api "github.com/hybrid-cloud-patterns/patterns-operator/api/v1alpha1"
-	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
-)
+	// Check if the config map exists and read the config map values
+	cm, err := r.CoreV1().ConfigMaps(OperatorNamespace).Get(context.Background(), OperatorConfigMap, metav1.GetOptions{})
+	// If we hit an error that is not related to the configmap not existing bubble it up
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
 
-const (
-	subscriptionNamespace = "openshift-operators"
-	applicationNamespace  = "openshift-gitops"
-)
```

Comment:
I know it's tempting to make related changes like this, but in the future (and particularly when dealing with upstream projects) I would advise against any change that isn't 100% necessary for the functionality under review.  It makes the PR longer, and increases the cognitive load for the reviewer who must mentally filter out those changes.

Fine this time, but these should normally be kept for a follow-up PR.

Reply:
Ok. Noted.